test/local_sensitivity/stiff_adjoints.jl

```diff
@@ -0,0 +1,212 @@
+end
+
+rdgrad = Zygote.gradient(loss_function,p)[1]
+@test fdgrad ≈ rdgrad rtol=1e-3
+
+loss_function = function(p)
+    prob = remake(prob0;u0=convert.(eltype(p),prob0.u0),p=p)
+    prediction = solve(prob, Rodas5(); saveat = 0.0:0.5:10.0,maxiters=maxiters,abstol=1e-8,reltol=1e-8)
+
+    tmpdata=prediction[[1,2],:];
+    tdata=target_data[[1,2],:];
+
+    # Calculate squared error
+    return sum(abs2,tmpdata-tdata)
+end
+
+rdgrad = Zygote.gradient(loss_function,p)[1]
+@test fdgrad ≈ rdgrad rtol=1e-3
+
+### OOP
+
+prob0_oop = ODEProblem{false}(lotka_volterra,u0,tspan,p0);
+# Solve the ODE and collect solutions at fixed intervals
+target_data = solve(prob0,RadauIIA5(), saveat =  0:0.5:10.0);
+
+loss_function = function(p)
+    prob = remake(prob0_oop;u0=convert.(eltype(p),prob0.u0),p=p)
+    prediction = solve(prob, RadauIIA5(); saveat = 0.0:0.5:10.0,maxiters=maxiters,abstol=1e-10,reltol=1e-10)
+
+    tmpdata=prediction[[1,2],:];
+    tdata=target_data[[1,2],:];
+
+    # Calculate squared error
+    return sum(abs2,tmpdata-tdata)
+end
+p=[1.5,1.2,1.4,1.6];
+
+fdgrad = ForwardDiff.gradient(loss_function,p)
+rdgrad = Zygote.gradient(loss_function,p)[1]
+
+@test fdgrad ≈ rdgrad rtol=1e-4
+
+loss_function = function(p)
+    prob = remake(prob0_oop;u0=convert.(eltype(p),prob0.u0),p=p)
+    prediction = solve(prob, TRBDF2(); saveat = 0.0:0.5:10.0,maxiters=maxiters,abstol=1e-10,reltol=1e-10)
+
+    tmpdata=prediction[[1,2],:];
+    tdata=target_data[[1,2],:];
+
+    # Calculate squared error
+    return sum(abs2,tmpdata-tdata)
+end
+
+rdgrad = Zygote.gradient(loss_function,p)[1]
+@test fdgrad ≈ rdgrad rtol=1e-3
+
+loss_function = function(p)
+    prob = remake(prob0_oop;u0=convert.(eltype(p),prob0.u0),p=p)
+    prediction = solve(prob, Rosenbrock23(); saveat = 0.0:0.5:10.0,maxiters=maxiters,abstol=1e-8,reltol=1e-8)
+
+    tmpdata=prediction[[1,2],:];
+    tdata=target_data[[1,2],:];
+
+    # Calculate squared error
+    return sum(abs2,tmpdata-tdata)
+end
+
+rdgrad = Zygote.gradient(loss_function,p)[1]
+@test fdgrad ≈ rdgrad rtol=1e-4
+
+loss_function = function(p)
+    prob = remake(prob0_oop;u0=convert.(eltype(p),prob0.u0),p=p)
+    prediction = solve(prob, Rodas5(); saveat = 0.0:0.5:10.0,maxiters=maxiters,abstol=1e-8,reltol=1e-8)
+
+    tmpdata=prediction[[1,2],:];
+    tdata=target_data[[1,2],:];
+
+    # Calculate squared error
+    return sum(abs2,tmpdata-tdata)
+end
+
+rdgrad = Zygote.gradient(loss_function,p)[1]
+@test fdgrad ≈ rdgrad rtol=1e-4
+
+# all implicit solvers
+solvers = [
+    # SDIRK Methods (ok)
+    ImplicitEuler(),
+    ImplicitMidpoint(),
+    Trapezoid(),
+    TRBDF2(),
+    SDIRK2(),
+    Kvaerno3(),
+    KenCarp3(),
+    Cash4(),
+    Hairer4(),
+    Hairer42(),
+    Kvaerno4(),
+    KenCarp4(),
+    Kvaerno5(),
+    KenCarp5(),
+    # Fully-Implicit Runge-Kutta Methods (FIRK)
+    RadauIIA5(),
+    # Fully-Implicit Runge-Kutta Methods (FIRK)
+    PDIRK44(),
+    # Rosenbrock Methods (all fail)

```

Comment:
Do Rosenbrock methods still fail after the fix?

Reply:
nope, they all work now.

Reply:
that is just copy-pasted from an issue.